src/main/java/io/kestra/plugin/typesense/Search.java

```diff
@@ -0,0 +1,137 @@
+package io.kestra.plugin.typesense;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.FileSerde;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.media.Schema.RequiredMode;
+import jakarta.validation.constraints.NotNull;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URI;
+import lombok.Builder;
+import lombok.Builder.Default;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+import org.typesense.api.Client;
+import org.typesense.model.SearchParameters;
+import org.typesense.model.SearchResult;
+import reactor.core.publisher.Flux;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Search documents",
+    description = "Search documents from a Typesense DB"
+)
+@Plugin(
+    examples = {
+        @io.kestra.core.models.annotations.Example(
+            title = "Search documents",
+            code = {
+                """
+                    id: typesense
+                    namespace: company.team
+
+                    tasks:
+                    - id: search
+                      type: io.kestra.plugin.typesense.Search
+                      apiKey: test-key
+                      port: 8108
+                      host: localhost
+                      collection: Countries
+                      query: Paris
+                      queryBy: capital
+                      filter: countryName: [France, England]
+                      sortBy: gdp:desc
+                    """
+            }
+        )
+    }
+)
+public class Search extends AbstractTypesenseTask implements RunnableTask<Search.Output> {
+
+    @Schema(
+        title = "The query",
+        example = "Tokyo",
+        requiredMode = RequiredMode.REQUIRED
+    )
+    @NotNull
+    protected Property<String> query;
+
+    @Schema(
+        title = "The fields to query",
+        example= "country, capital",
+        requiredMode = RequiredMode.REQUIRED
+    )
+    @NotNull
+    protected Property<String> queryBy;
+
+    @Schema(
+        title = "The filters to apply to the query",
+        example = "countryName: [France, Germany]"
+    )
+    @Default
+    protected Property<String> filter = Property.of("");
```

Comment:
default of empty seems not very usefull, maybe add no default as a property is always render via an Optional